storybookjs/create-svelte-with-args
# Initial package

This PR sets up the initial functionality for a package that takes CLI arguments and passes them along to `create-svelte` to create a SvelteKit project.

One big decision to make here, is whether we want the wrapper to be explicit or implicit. This implementation is explicit, meaning that only the arguments actually specified are supported. The upside of this is that the CLI is documented nicely. The downside is that it makes it a bit more tedious to keep this package up to date when upstream `create-svelte` is changed. It's not _that_ tedious considering each argument is 4 LOC, but it probably means we can't auto upgrade it without manual interference.

The alternative is the implicit angle, where we just pass along whatever arguments are provided to the CLI directly to the `create-svelte`. This way we could have a script that automatically keeps this package up to date because it doesn't really do anything:
1. Update the `create-svelte` dependency
2. Set the new version of this package to match `create-svelte`

Documentation would be a little more flawed since we'd basically have to point users towards [`create-svelte`](https://github.com/sveltejs/kit/tree/master/packages/create-svelte) and say "any CLI args you pass translates 1-to-1 with arguments required in `create()`.

Thoughts @ndelangen ?

The CLI outputs the following from the `--help` command:

```bash
$ node ./dist/bin.js --help
create-svelte-with-args [args]

Options:
      --help        Show help                                          [boolean]
      --version     Show version number                                [boolean]
  -n, --name        The name of the project                  [string] [required]
  -d, --directory   The directory to install the project in. Defaults to the
                    project name                                        [string]
  -t, --template    The template to use
             [string] [required] [choices: "default", "skeleton", "skeletonlib"]
  -y, --types       How types will be written
                  [string] [required] [choices: "checkjs", "typescript", "null"]
  -p, --prettier    Whether prettier should be included     [boolean] [required]
  -e, --eslint      Whether eslint should be included       [boolean] [required]
  -l, --playwright  Whether playwright should be included   [boolean] [required]
      --dry         Whether to run in dry run mode, not creating any files
                                                      [boolean] [default: false]

Examples:
  create-svelte-with-args                   Default app, checking JS types, with
  --name=my-new-app --template=default      Prettier, and no ESLint or
  --types=checkjs --prettier --no-eslint    Playwright
  --no-playwright
  create-svelte-with-args                   Default app, installing in specified
  --name=my-new-app                         directory
  --directory=other-dir/my-app
  --template=default --types=typescript
  --no-prettier --no-eslint
  --no-playwright

For more information, see https://github.com/storybookjs/create-svelte-with-args
```

## How to test

1. Checkout
2. `yarn install`
3. `yarn start --name=my-app --template=default --types=checkjs --no-prettier --no-eslint --no-playwright`
4. See the scaffolded SvelteKit project in `./my-app`

## Todo in later PRs

- [ ] Proper documentation in README
- [ ] ~~Follow same versioning as upstream `create-svelte`~~
- [x] Set up publishing #2 
- [x] GH action to run checks (and publish?) #2
- [ ] ~~(maybe) ESLint. Ran into trouble when extending the Storybook config. Might not be worth it for such a small project.~~
